Crossgen2.win-x64 symbols nupkg invalid

When adding symbols validation to the new release pipeline, we found that Microsoft.NETCore.App.Crossgen2.win-x64.5.0.0-preview.2.20160.6.symbols.nupkg incorrectly has 2 copies of System.Private.CoreLib.ni.dll, so when symbols-validation tries to extract the package, we get the following error:

```
Validating Microsoft.NETCore.App.Crossgen2.win-x64.5.0.0-preview.2.20160.6.symbols.nupkg 
at CountMissingSymbols, D:\a\1\s\eng\common\post-build\symbols-validation.ps1: line 84
at <ScriptBlock>, D:\a\1\s\eng\common\post-build\symbols-validation.ps1: line 150
at CheckSymbolsAvailable, D:\a\1\s\eng\common\post-build\symbols-validation.ps1: line 131
at <ScriptBlock>, D:\a\1\s\eng\common\post-build\symbols-validation.ps1: line 195
at <ScriptBlock>, D:\a\_temp\0f6d2992-7121-4785-bf19-39088c4126cc.ps1: line 2
at <ScriptBlock>, <No file>: line 1
##[error](NETCORE_ENGINEERING_TELEMETRY=CheckSymbols) Exception calling "ExtractToDirectory" with "2" argument(s): "The file 'D:\a\1\a\symbols\16be524d-17a5-46e9-92b4-afd9ee901962\tools\System.Private.CoreLib.ni.pdb' already exists."
```

The failing build is here: https://dev.azure.com/dnceng/internal/_build?definitionId=799&_a=summary

To repro, pull down Microsoft.NETCore.App.Crossgen2.win-x64.5.0.0-preview.2.20160.6.symbols.nupkg from the above build artifacts, and try to extract it using [System.IO.Compression.ZipFile]::ExtractToDirectory using powershell, or change the nupkg to a zip and open it in a file browser where you will see 2 copies of System.Private.CoreLib.ni.pdb.

--- a/eng/liveBuilds.targets
+++ b/eng/liveBuilds.targets
@@ -86,11 +86,13 @@
 
     <PropertyGroup Condition="'$(RuntimeFlavor)' == 'CoreCLR'">
       <CoreCLRArtifactsPath>$([MSBuild]::NormalizeDirectory('$(CoreCLRArtifactsPath)'))</CoreCLRArtifactsPath>
+      <CoreCLRArtifactsPdbDir>$([MSBuild]::NormalizeDirectory('$(CoreCLRArtifactsPath)','PDB'))</CoreCLRArtifactsPdbDir>
       <!--
         Even though CoreCLRSharedFrameworkDir is statically initialized, set it again in case the
         value is different after CoreCLRArtifactsPath is normalized.
       -->
       <CoreCLRSharedFrameworkDir>$([MSBuild]::NormalizeDirectory('$(CoreCLRArtifactsPath)','sharedFramework'))</CoreCLRSharedFrameworkDir>
+      <CoreCLRSharedFrameworkPdbDir>$([MSBuild]::NormalizeDirectory('$(CoreCLRSharedFrameworkDir)','PDB'))</CoreCLRSharedFrameworkPdbDir>
       <CoreCLRCrossTargetComponentDir
         Condition="'$(CoreCLRCrossTargetComponentDirName)' != ''">$([MSBuild]::NormalizeDirectory('$(CoreCLRArtifactsPath)','$(CoreCLRCrossTargetComponentDirName)','sharedFramework'))</CoreCLRCrossTargetComponentDir>
     </PropertyGroup>
@@ -112,18 +114,19 @@
       <RuntimeFiles Include="@(_systemPrivateCoreLib)" />
       <RuntimeFiles
         Include="
-          $(CoreCLRSharedFrameworkDir)PDB/*.pdb;
-          $(CoreCLRSharedFrameworkDir)PDB/*.dbg;
-          $(CoreCLRSharedFrameworkDir)PDB/*.dwarf" />
+          $(CoreCLRSharedFrameworkPdbDir)*.pdb;
+          $(CoreCLRSharedFrameworkPdbDir)*.dbg;
+          $(CoreCLRSharedFrameworkPdbDir)*.dwarf" />
       <RuntimeFiles
-        Include="$(CoreCLRArtifactsPath)PDB/System.Private.CoreLib.pdb;" />
-      <RuntimeFiles Condition="Exists('$(CoreCLRArtifactsPath)PDB/System.Private.CoreLib.ni.pdb')"
-        Include="$(CoreCLRArtifactsPath)PDB/System.Private.CoreLib.ni.pdb" />
+        Include="$(CoreCLRArtifactsPdbDir)System.Private.CoreLib.pdb" />
+      <RuntimeFiles Condition="Exists('$(CoreCLRArtifactsPdbDir)System.Private.CoreLib.ni.pdb')"
+        Include="$(CoreCLRArtifactsPdbDir)System.Private.CoreLib.ni.pdb" />
+
       <CoreCLRCrossTargetFiles Condition="'$(CoreCLRCrossTargetComponentDir)' != ''"
         Include="
-          $(CoreCLRCrossTargetComponentDir)PDB/*.pdb;
-          $(CoreCLRCrossTargetComponentDir)PDB/*.dbg;
-          $(CoreCLRCrossTargetComponentDir)PDB/*.dwarf" />
+          $(CoreCLRSharedFrameworkPdbDir)*.pdb;
+          $(CoreCLRSharedFrameworkPdbDir)*.dbg;
+          $(CoreCLRSharedFrameworkPdbDir)*.dwarf" />
 
       <CoreCLRCrossTargetFiles>
         <TargetPath>runtime/$(CoreCLRCrossTargetComponentDirName)_$(TargetArchitecture)/native</TargetPath>

--- a/src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj
+++ b/src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj
@@ -87,7 +87,11 @@
     </MSBuild>
 
     <ItemGroup>
-      <File Include="@(Crossgen2File)">
+      <!-- Avoid having two copies of System.Private.CoreLib.ni.pdb in the same directory -->
+      <Crossgen2File Update="@(Crossgen2File)" SavedIdentity="%(Identity)" />
+      <Crossgen2FileDistinct Include="@(Crossgen2File->'%(Filename)%(Extension)'->Distinct())" />
+
+      <File Include="@(Crossgen2FileDistinct->'%(SavedIdentity)')" RemoveMetadata="SavedIdentity">
         <TargetPath>tools</TargetPath>
       </File>
     </ItemGroup>

